File: 3171-minimum-equal-sum-of-two-arrays-after-replacing-zeros/3171-minimum-equal-sum-of-two-arrays-after-replacing-zeros.cpp
class Solution {
public:
    long long minSum(vector<int>& nums1, vector<int>& nums2) {
        long long sum1 = 0, sum2 = 0, zeros1 = 0, zeros2 = 0;
        for(auto it : nums1) {
            sum1 += it;
            if(it == 0) zeros1++;
        }
        for(auto it : nums2) {
            sum2 += it;
            if(it == 0) zeros2++;
        }

        if(zeros1 == 0 && sum1 < sum2 + zeros2) return -1;
        if(zeros2 == 0 && sum2 < sum1 + zeros1) return -1;

        return max(sum1 + zeros1, sum2 + zeros2);
    }
};

File: 3171-minimum-equal-sum-of-two-arrays-after-replacing-zeros/3171-minimum-equal-sum-of-two-arrays-after-replacing-zeros.java
class Solution {
    public long minSum(int[] nums1, int[] nums2) {
        long s1 = 0, s2 = 0, c1 = 0, c2 = 0;
        for(int it : nums1) {
            s1 += it;
            if(it == 0) c1++;
        }

        for(int it : nums2) {
            s2 += it;
            if(it == 0) c2++;
        }

        if(c1 == 0 && s1 < s2 + c2) return -1;
        if(c2 == 0 && s2 < s1 + c1) return -1;
        return Math.max(c1 + s1, s2 + c2);
    }
}